pkg/compiler/codegen.go

```diff
@@ -2011,28 +2011,41 @@ func (c *codegen) writeJumps(b []byte) ([]byte, error) {
 			}
 		}
 	}
+
+	if c.deployEndOffset >= 0 {
+		_, end := correctRange(uint16(c.initEndOffset+1), uint16(c.deployEndOffset), offsets)
+		c.deployEndOffset = int(end)
```

Comment:
Why do you adjust `deploy` first and then `init`? Shouldn't it be the other way around (going through the script sequentially)?

Reply:
Or it probably doesn't really matter here.

Reply:
At first it was so because `correctRange` expects `initEndOffset+1 = deployStartOffset` to be unchanged. However we don't really care use start offset here, so yes, it doesn't matter.